gensim/similarities/docsim.py

```diff
@@ -392,12 +392,14 @@ def __iter__(self):
         # turn off query normalization (vectors in the index are already normalized, save some CPU)
         norm, self.norm = self.norm, False
 
-        for chunk in self.iter_chunks():
-            if chunk.shape[0] > 1:
-                for sim in self[chunk]:
-                    yield sim
-            else:
-                yield self[chunk]
+        import multiprocessing
+        import functools
+        pool = multiprocessing.Pool()
+        worker = functools.partial(_query_chunk_worker, index=self)
+        for result in pool.map(worker, self.iter_chunks()):
```

Comment:
This will be problematic: multiprocessing has no queue limits, it keeps feeding the input queue constantly. This will blow up the memory (queue up all chunks), for large enough index, slow enough workers or slow enough result consumption after `yield`.